janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java

```diff
@@ -252,21 +252,25 @@ Cluster initializeCluster() throws PermanentBackendException {
     }
 
     Session initializeSession(final String keyspaceName) {
-        final Configuration configuration = getStorageConfig();
-        final Map<String, Object> replication = Match(configuration.get(REPLICATION_STRATEGY)).of(
+        final Session s = this.cluster.connect();
+
+        if (this.cluster.getMetadata().getKeyspace(keyspaceName) == null) {
+            final Configuration configuration = getStorageConfig();
+            final Map<String, Object> replication = Match(configuration.get(REPLICATION_STRATEGY)).of(
                 Case($("SimpleStrategy"), strategy -> HashMap.<String, Object> of("class", strategy, "replication_factor", configuration.get(REPLICATION_FACTOR))),
```

Comment:
add a comment on what is going on with the `$()`

Reply:
hmmm, I just bit hesitate to add comments on this bit as am not the author of this piece of code. I can add comments on it based on my understanding but need reviews for it though..